From: Vadim Zhukov <persgray@gmail.com>
Date: Sat, 25 May 2013 14:19:24 +0000 (+0100)
Subject: pidl: Recent Perl warns about "defined(@var)" constructs.
X-Git-Tag: tdb-1.2.12~99
X-Git-Url: https://git.samba.org/?p=samba.git;a=commitdiff_plain;h=92254d09e0ee5a7d9d0cd91fe1803f54e64d9a5f

pidl: Recent Perl warns about "defined(@var)" constructs.

Signed-off-by: Jelmer Vernooij <jelmer@samba.org>

Autobuild-User(master): Jelmer Vernooij <jelmer@samba.org>
Autobuild-Date(master): Sat May 25 18:10:53 CEST 2013 on sn-devel-104
---

diff --git a/pidl/lib/Parse/Pidl/ODL.pm b/pidl/lib/Parse/Pidl/ODL.pm
index 74d9ac7..14e77fa 100644
--- a/pidl/lib/Parse/Pidl/ODL.pm
+++ b/pidl/lib/Parse/Pidl/ODL.pm
@@ -70,7 +70,7 @@ sub ODL2IDL
 					next;
 				}
 				my $podl = Parse::Pidl::IDL::parse_file($idl_path, $opt_incdirs);
-				if (defined(@$podl)) {
+				if (defined($podl)) {
 					require Parse::Pidl::Typelist;
 					my $basename = basename($idl_path, ".idl");
 
diff --git a/pidl/pidl b/pidl/pidl
index 2a46e92..c65092e 100755
--- a/pidl/pidl
+++ b/pidl/pidl
@@ -605,7 +605,7 @@ sub process_file($)
 		require Parse::Pidl::IDL;
 
 		$pidl = Parse::Pidl::IDL::parse_file($idl_file, \@opt_incdirs);
-		defined @$pidl || die "Failed to parse $idl_file";
+		defined $pidl || die "Failed to parse $idl_file";
 	}
 
 	require Parse::Pidl::Typelist;
